crates/oxc_parser/src/lexer/trivia_builder.rs

```diff
@@ -201,6 +201,71 @@ impl TriviaBuilder {
     }
 }
 
+#[inline(always)]
+fn contains_license_or_preserve_comment(s: &str) -> bool {
+    use memchr::memchr_iter;
```

Comment:
https://docs.rs/memchr/latest/memchr/index.html#example-repeating-a-search-for-the-same-needle Probably need a finder.

Reply:
hmm i tried a lazy lock, and putting the finder in the TriviaBuilder struct. both seem to make perf slower than using memchr iter 🤔

Reply:
hmm using a finder both via once lock/ on `TrivaBuilder` seemed to negate any perf improvements